gunicorn_thrift/gevent_worker.py

```diff
@@ -45,6 +45,10 @@ def handle(self, listener, client, addr):
 
             try:
                 while True:
+                    self.nr += 1
+                    if self.alive and self.nr >= self.max_requests:
+                        self.log.info("Autorestarting worker after current process.")
+                        self.alive = False                    
```

Comment:
`while self.slave` can make it stop, but the client connected would raise a exception before worker stop.

https://github.com/benoitc/gunicorn/commit/4ed990170829cc708b5eaa03e14de10140b25109